lib/haproxy/parser.rb

```diff
@@ -11,7 +11,32 @@ class Parser
     }
     # Added in haproxy 1.4
     SERVER_ATTRIBUTE_NAMES_1_4 = %w{error-limit observe on-error}
-    SERVER_ATTRIBUTE_NAMES = SERVER_ATTRIBUTE_NAMES_1_3 + SERVER_ATTRIBUTE_NAMES_1_4
+    # Added in haproxy 1.5
+    SERVER_ATTRIBUTE_NAMES_1_5 = %w{ 
+      agent-check agent-inter agent-port ca-file check-send-proxy check-ssl
+      ciphers crl-file crt error-limit force-sslv3 force-tlsv10 force-tlsv11
+      force-tlsv12 no-sslv3 no-tls-tickets no-tlsv10 no-tlsv11 no-tlsv12 observe
+      on-error on-marked-down on-marked-up send-proxy send-proxy-v2 send-proxy-v2-ssl
+      send-proxy-v2-ssl-cn ssl verify verifyhost
+    }
+    # Added in haproxy 1.6
+    SERVER_ATTRIBUTE_NAMES_1_6 = %w{
+      namespace no-ssl-reuse non-stick resolve-prefer resolvers sni tcp-ut
+    }
+    # Added in haproxy 1.7
+    SERVER_ATTRIBUTE_NAMES_1_7 = %w{
+      agent-send init-addr resolve-net
+    }
+    SERVER_ATTRIBUTE_NAMES_1_8 = %w{
+      agent-addr check-sni enabled force-tlsv13 no-agent-check no-backup no-check no-check-ssl
+      no-send-proxy no-send-proxy-v2 no-ssl no-tlsv13 no-verifyhost ssl-max-ver ssl-min-ver
+      ssl-reuse stick tls-tickets
+    }
+    SERVER_ATTRIBUTE_NAMES_1_9 = %w{
+      no-send-proxy-v2-ssl no-send-proxy-v2-cn
```

Comment:
Looking at the docs, I think `no-send-proxy-v2-cn` should be `no-send-proxy-v2-ssl-cn`.

Also, both of these attributes appear to be available in 1.8 as well. (Although it's a bit academic, since we're just mashing them together into a single array at the moment.)

Reply:
I reviewed the documentation and found a couple other attributes in the wrong location.  I agree the version variables are a bit academic, but coming from an IT background, the sometimes the little bits of self documentation can help to resolve complex issues.  I believe as long as it doesn't introduce errors, it's a good thing. 

This merge would be the greatest opportunity to introduce errors considering I added 4 versions of haproxy attributes.  As long as these attributes are accurate hopefully it will make the next version easier.

Reply:
I agree that having them grouped is useful as documentation. I appreciate you going back and reviewing them.

IMO, adding these new attributes does more good than harm, since the gem doesn't support targeting a specific version anyway.

👍